pandas/core/frame.py

```diff
@@ -4949,6 +4949,14 @@ def _combine_match_columns(self, other, func, level=None, try_cast=True):
         return self._constructor(new_data)
 
     def _combine_const(self, other, func, errors='raise', try_cast=True):
+        if lib.is_scalar(other) or np.ndim(other) == 0:
```

Comment:
is is pretty annoything that we have to do this, I would make an explict function maybe ``is_any_scalar`` I think as we have these types of checks all over. pls make an issue for this.

Reply:
Done.